index.js

```diff
@@ -1,236 +1,349 @@
+/**
+ * Top-level s3-asset-uploader module
+ * @module s3-asset-uploader
+ * @see README.md
+ */
+
+// Node imports
 const fs = require('fs')
-const crypto = require('crypto')
 const path = require('path')
-const finder = require('findit')
-const debug = require('debug')('mix:s3')
-const Promise = require('bluebird')
+// NPM imports
+const debug = require('debug')('s3-asset-uploader')
 const AWS = require('aws-sdk')
-const mime = require('mime')
+const Bluebird = require('bluebird')
+// Lib imports
+const directoryLib = require('./lib/directory')
+const fileLib = require('./lib/file')
+const hashLib = require('./lib/hash')
+const transformLib = require('./lib/transform')
 
-const DEFAULT_HEADERS = {
-  'ACL': 'public-read'
-}
-const TYPES = {
-  'js': 'utf8',
-  'css': 'utf8',
-  'sass': 'utf8',
-  'scss': 'utf8',
-  'png': 'binary',
-  'jpg': 'binary',
-  'jpeg': 'binary',
-  'gif': 'binary',
-  'ico': 'binary'
+const FILE_EXTENSION_REGEXP = /((\.\w+)?\.\w+)$/
+const DEFAULT_ACL = 'public-read'
+const DEFAULT_DIGEST_FILE_NAME = 'asset-map.json'
+const DEFAULT_GZIP_CACHE_CONTROL = 'max-age=1314000'
```

Comment:
To make this more readable, you might want to convert this to 
```js
`max-age=${365*24*60*60}`
```

Either it needs to be altered because the math is actually wrong for this. 😕
1314000 == 365*60*60 which is a nonsensical time.
I believe max-age should be in seconds.

Reply:
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html#expiration-individual-objects

Reply:
Good catch; I borrowed that value from [the existing implementation](https://github.com/mix/s3-asset-uploader/blob/92b11146c406168bb6de302a9dd3aca18893ccba/index.js#L219) 😬